src/components/tooltip/tooltip.pw.tsx

```diff
@@ -0,0 +1,460 @@
+import React from "react";
+import { test, expect } from "@playwright/experimental-ct-react17";
+import { Locator } from "@playwright/test";
+import {
+  TooltipComponent,
+  UncontrolledTooltipComponent,
+  Default,
+  Controlled,
+  Positioning,
+  FlipBehaviourOverrides,
+  LargeTooltip,
+  Types,
+  ColorOverrides,
+  TooltipWithChangingTargetComponent,
+} from "./components.test-pw";
+import {
+  checkAccessibility,
+  assertCssValueIsApproximately,
+} from "../../../playwright/support/helper";
+import { SIZE, COLOR, CHARACTERS } from "../../../playwright/support/constants";
+import { TooltipProps } from "../../../src/components/tooltip/tooltip.component";
+import { getDataElementByValue } from "../../../playwright/components";
+
+const backgroundColors = [COLOR.ORANGE, COLOR.RED, COLOR.BLACK, COLOR.BROWN];
+const testData = [CHARACTERS.DIACRITICS, CHARACTERS.SPECIALCHARACTERS];
+
+test.describe("Tooltip component", () => {
+  testData.forEach((message) => {
+    test(`should render when message prop is set to ${message}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<TooltipComponent message={message} />);
+
+      await expect(getDataElementByValue(page, "tooltip")).toHaveText(message);
+    });
+  });
+
+  testData.forEach((id) => {
+    test(`should render when id prop is set to ${id}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<TooltipComponent id={id} />);
+
+      await expect(getDataElementByValue(page, "tooltip")).toHaveId(id);
+    });
+  });
+
+  [true, false].forEach((boolVal) => {
+    test(`should render when tooltip visibility is set to ${boolVal}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<TooltipComponent isVisible={boolVal} />);
+
+      if (boolVal === true) {
+        await expect(getDataElementByValue(page, "tooltip")).toBeVisible();
+      } else {
+        await expect(getDataElementByValue(page, "tooltip")).not.toBeVisible();
+      }
+    });
+  });
```

Comment:
**suggestion**: I would avoid using if blocks like this where possible and instead create two independent tests for the visible and non-visible cases, as this will improve the test's readability and error messaging if they happen to fail.

```suggestion
  test("tooltip should be visible when isVisible prop is true", async ({ mount, page }) => {
    await mount(<TooltipComponent isVisible />);
    await expect(getDataElementByValue(page, "tooltip")).toBeVisible();
  });

  test("tooltip should not be visible when isVisible prop is false", async ({ mount, page }) => {
    await mount(<TooltipComponent isVisible={false} />);
    await expect(getDataElementByValue(page, "tooltip")).not.toBeVisible();
  });
```